openupgradelib/openupgrade_merge_records.py

```diff
@@ -438,6 +445,11 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
                 if not getattr(target_record, field.name) and zip_list:
                     vals[field.name] = zip_list[0][0]
                     vals[field.model_field] = zip_list[0][1]
+        elif field.type == 'reference':
```

Comment:
This is the exact same code as `many2one`, so include both in the same if:
`elif field.type in ('many2one', 'reference'):`